### ISSUE TITLE
coreos-installer fails when installation path is set to a path below /dev/disk

### ISSUE THREAD

**Issue by elemental-lf** [coreos/coreos-installer#242]
# Bug #

The `coreos-installer` fails when the installation disk is specified by using a path under `/dev/disk` (like `/dev/disk/by-path/pci-0000:5c:00.0-scsi-0:1:0:0`). It works when using a direct reference to a device directly under `/dev` (like `/dev/sdi`).  The bug only occurs when the `--ignition` command line option is used.

## Host Operating System Version ##

Fedora CoreOS 31 (2247.6.0)

* `fedora-coreos-31.20200407.3.0-live-kernel-x86_64`
* `fedora-coreos-31.20200407.3.0-live-initramfs.x86_64.img`
* `fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz`

## Target Operating System Version ##

Same as above.

## coreos-installer Version ##

`0.1.3`

## Expected Behavior ##

The installation should complete without error.

## Actual Behavior ##

The installation fails because it is trying to mount a wrongly constructed device path (`/dev/disk/by-path/sdi1`):

```
         Starting CoreOS Installer...
[  311.244897] audit: type=1131 audit(1589824254.991:88): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[  311.321800]  sdi: sdi1 sdi2 sdi3 sdi4 sdi6 sdi7 sdi9
[  278.746972] coreos-installer-service[1651]: coreos-installer install /dev/disk/by-path/pci-0000:5c:00.0-scsi-0:1:0:0 --ignition /tmp/coreos-installer-xwQhBX --firstboot-args rd.neednet=1 ip=dhcp  --image-url http://<redacted>:8080/assets/fedora-coreos/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz
[  279.397191] coreos-installer-service[1651]: Downloading image from http://<redacted>:8080/assets/fedora-coreos/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz
[  279.592944] coreos-installer-service[1651]: Downloading signature from http://<redacted>:8080/assets/fedora-coreos/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz.sig
[  280.130234] coreos-installer-service[1651]: Read disk 29.8 MiB/461.6 MiB (6%)
[...]
[  348.480915] coreos-installer-service[1651]: Read disk 461.6 MiB/461.6 MiB (100%)
[  349.009502] coreos-installer-service[1651]: gpg: Signature made Tue Apr 21 21:20:22 2020 UTC
[  349.111943] coreos-installer-service[1651]: gpg:                using RSA key 50CB390B3C3359C4
[  349.217948] coreos-installer-service[1651]: gpg: Good signature from "Fedora (31) <fedora-31-primary@fedoraproject.org>" [ultimate]
[  349.362954] coreos-installer-service[1651]: R[  381.930730] GPT:Primary header thinks Alt. header is not at the end of the disk.
ead disk 461.6 M[  382.022701] GPT:5593087 != 781357231
iB/461.6 MiB (10[  382.082456] GPT:Alternate GPT header not at the end of the disk.
0%)
[  382.171585] GPT:5593087 != 781357231
[  382.219858] GPT: Use GNU Parted to correct GPT errors.
[  382.282984]  sdi: sdi1 sdi2 sdi3 sdi4
[  350.033832] coreos-installer-service[1651]: Error: mounting device /dev/disk/by-path/sdi1 on /tmp/coreos-installer-1lpvQG
[  350.167959] coreos-installer-service[1651]: Caused by: ENOENT: No such file or directory
[FAILED] Failed to start CoreOS Installer.
See 'systemctl status coreos-installer.service' for details.
[  350.265956] coreos-installer-service[1651]: Clearing partition table
[DEPEND] Dependency failed for CoreOS Installer Target.
[  350.483057] c[  383.097313] audit: type=1130 audit(1589824326.843:89): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=coreos-installer comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
oreos-installer-service[1651]: Error: install failed
```

## Reproduction Steps ##

```
# ls -al /dev/disk/by-path/pci-0000:01:04.0-scsi-0:0:0:3
lrwxrwxrwx. 1 root root 9 19. Mai 15:15 /dev/disk/by-path/pci-0000:01:04.0-scsi-0:0:0:3 -> ../../sdd
```

```
# coreos-installer install /dev/disk/by-path/pci-0000:01:04.0-scsi-0:0:0:3 --image-file /home/core/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz  --ignition /dev/null
Copying image from /home/core/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz
Reading signature from /home/core/fedora-coreos-31.20200407.3.0-metal.x86_64.raw.xz.sig
gpg: Signature made Di 21 Apr 2020 21:20:22 UTC
gpg:                using RSA key 50CB390B3C3359C4
gpg: Good signature from "Fedora (31) <fedora-31-primary@fedoraproject.org>" [ultimate]
> Read disk 461.6 MiB/461.6 MiB (100%)   
Error: mounting device /dev/disk/by-path/sdd1 on /tmp/coreos-installer-53spVB
Caused by: ENOENT: No such file or directory
Clearing partition table
Error: install failed

```

## Other Information ##

We're migrating from Container Linux to FCOS and this path specification has worked with the CL installer. Unfortunately the logical RAID-1 volume on these HP servers which is used as the system disk is enumerated last by the Linux kernel and so any changes of the storage configuration lead to a shift of this device name. So simply specifying `/dev/sdi`  instead of `/dev/disk/by-path/pci-0000:5c:00.0-scsi-0:1:0:0` is not an option as it might lead to a destruction of a data disk when the `coreos-installer` installs an OS on it.

**Comment by nikita-dubrovskii** [comment]
I guess this issue is fixed in newer versions of installer:
https://github.com/coreos/coreos-installer/pull/215

**Comment by jlebon** [comment]
Yup, this is fixed by #215 (see https://github.com/coreos/coreos-installer/pull/215#issuecomment-618648076), which is in 0.2.0, which is the latest stable now (31.20200505.3.0). Could you try again with that?

**Comment by dustymabe** [comment]
I'll mark this as closed, fixed by #215. @elemental-lf - can you try and report back on success/failure?
